src/module-elasticsuite-indices/Controller/Adminhtml/Analysis/Request.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade Smile ElasticSuite to newer
+ * versions in the future.
+ *
+ * @category  Smile
+ * @package   Smile\ElasticsuiteIndices
+ * @author    Vadym HONCHARUK <vahonc@smile.fr>
+ * @copyright 2022 Smile
+ * @license   Open Software License ("OSL") v. 3.0
+ */
+
+namespace Smile\ElasticsuiteIndices\Controller\Adminhtml\Analysis;
+
+use Magento\Backend\App\Action;
+use Magento\Backend\App\Action\Context;
+use Magento\Framework\Controller\Result\JsonFactory;
+use Smile\ElasticsuiteCore\Api\Client\ClientInterface;
+
+/**
+ * ElasticSuite Indices Adminhtml Analysis Request Controller.
+ *
+ * @category Smile
+ * @package  Smile\ElasticsuiteIndices
+ * @author   Vadym HONCHARUK <vahonc@smile.fr>
+ */
+class Request extends Action
+{
+    /**
+     * @var JsonFactory
+     */
+    protected $resultJsonFactory;
+
+    /**
+     * @var ClientInterface
+     */
+    private $client;
+
+    /**
+     * @param ClientInterface $client
+     * @param JsonFactory     $resultJsonFactory
+     * @param Context         $context
+     *
+     */
+    public function __construct(
+        ClientInterface $client,
+        JsonFactory $resultJsonFactory,
+        Context $context
+    )
+    {
+        $this->client = $client;
+        $this->resultJsonFactory = $resultJsonFactory;
+
+        return parent::__construct($context);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute()
+    {
+        $errors = [];
+
+        if ($_POST['index'] == '0') {
```

Comment:
$_POST['index'] should be replaced by $this->getRequest()->getPost($index, null);

Same for other posted variables.